src/internal/streamstore/streamstore.go

```diff
@@ -99,3 +104,143 @@ func (di *streamItem) ToReader() io.ReadCloser {
 func (di *streamItem) Deleted() bool {
 	return false
 }
+
+// ---------------------------------------------------------------------------
+// common reader/writer/deleter
+// ---------------------------------------------------------------------------
+
+type backuper interface {
```

Comment:
nit: we should probably define this in a single common location instead of ad-hoc in multiple files (maybe kopia package? Not sure if that would create cycles later on)

Reply:
I'll make a ticket for that and leave it for tech debt cleanup.  We can focus on getting the best locations, names, etc for it therein.